joomla/joomla-cms
# Fix database error after new installation of current staging

Pull Request for Issue #25293 .

### Summary of Changes

This PR fixes 2 problems coming from PR #24595 :

1. The database schema checker can't handle removing and adding back an index with the same name. Since that index was not useful anyway, this PR changes the schema updates from PR #24595 and the installation joomla.sql so this index is not added back, but 2 useful indexes are added istead. For details on these changes see [https://github.com/joomla/joomla-cms/issues/25293#issuecomment-504742103](https://github.com/joomla/joomla-cms/issues/25293#issuecomment-504742103).
2. This PR also fixes the typo `ALTER TABLE "#__template_style"` instead of `ALTER TABLE "#__template_styles"` in the schema update from PR #24595 for postgresql.

### Testing Instructions

1. Install current staging and check database view in backend. Result: See section "Actual result" below.
2. Use "fix" button: Result: does not help. You will end in a toggling of database error "inxed should exist" and "index should not exist" with every click on the "fix" button.
3. Apply patch of this PR e.g. with Patchtester component.
4. Refresh database view. Result three errors: 1. index `idx_home` should not exist, 2. `idx_client_id` should exist, and 3. `idx_client_id_home` should exist. 
5. Use "fix" button. Result: See section "Expected result" below.
6. Remove configuration.php and clear database content e.g. with MyPhpAdmin and repeat clean installation. Result: See section "Expected result" below.

For Postgresql the number of errors shown in step 4 and for the actual result is different because of the typo in the schema update for postgresql without this PR, but at the end the result will be the same: Fix button in step 5 fixes it, and after new install with this PR applied there are no errors.
### Expected result

No database errors.

### Actual result

1 error: `Table 'xyz_template_styles' should not have index 'idx_home'. (From file 3.9.8-2019-06-15.sql.)`

### Documentation Changes Required

None.
